vllm/v1/attention/backends/flex_attention.py

```diff
@@ -0,0 +1,467 @@
+# SPDX-License-Identifier: Apache-2.0
+"""Attention layer with FlashAttention."""
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Optional
+
+import torch
+from attn_gym.masks.document_mask import _offsets_to_doc_ids_tensor
```

Comment:
`attn_gym` is not a dependency of vLLM atm.
Given that `_offsets_to_doc_ids_tensor` is a simple helper function, can we simply copy the code into this file?

Reply:
Ohh great catch I thought I removed this already